src/main/java/com/shopping_app/shoppingApp/domain/BaseEntity.java

```diff
@@ -0,0 +1,39 @@
+package com.shopping_app.shoppingApp.domain;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@MappedSuperclass
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public abstract class BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    protected Long id;
+
+    @Column(name = "created_at", updatable = false, nullable = false)
+    protected LocalDateTime createAt;
+
+    @Column(name = "updated_at", updatable = true, nullable = false)
+    protected LocalDateTime updatedAt;
+
+    @PrePersist
+    protected void onCreate() {
+        this.createAt = this.updatedAt = (this.createAt == null ? LocalDateTime.now() : this.createAt);
+    }
+
+    @PreUpdate
+    protected void onUpdate() {
+        this.updatedAt = LocalDateTime.now();
+    }
```

Comment:
There is a better way to do this, can you see if there are any annotation available for storing the create/update timestamp?

Reply:
Ohh Okk